### PR TITLE
Don't run sql precheck if projects.get call fails

### PR DESCRIPTION
**Pull request by rileykarson**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/7595

Reproduced locally, and this fixes the error the user experienced. This requires a whole bunch of host project / service project IAM setup that doesn't make it worth setting up an integration test for imo.

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
sql: fixed a case in `google_sql_database_instance` where we inadvertently required the `projects.get` permission for a service networking precheck introduced in `v3.44.0`
```

### CODE REVIEW

File: third_party/terraform/resources/resource_sql_database_instance.go.erb [hunk 1]
```diff
@@ -664,22 +664,30 @@ func resourceSqlDatabaseInstanceCreate(d *schema.ResourceData, meta interface{})
...
 		serviceNetworkingNetworkName, err := retrieveServiceNetworkingNetworkName(d, config, network, userAgent)
 		if err != nil {
-			return err
-		}
-		response, err := config.NewServiceNetworkingClient(userAgent).Services.Connections.List("services/servicenetworking.googleapis.com").
-			Network(serviceNetworkingNetworkName).Do()
-		if err != nil {
-			// It is possible that the identity creating the SQL Instance might not have permissions to call servicenetworking.services.connections.list
-			log.Printf("[WARNING] Failed to list Service Networking of the project. Skipping Service Networking check.")
+			var gerr *googleapi.Error
```
Comment (ScottSuarez) [inline third_party/terraform/resources/resource_sql_database_instance.go.erb:676]:
```suggestion
			
			// googleapi.Error is a permission error for the precheck
			var gerr *googleapi.Error
```

File: third_party/terraform/resources/resource_sql_database_instance.go.erb [hunk 2]
```diff
@@ -664,22 +664,30 @@ func resourceSqlDatabaseInstanceCreate(d *schema.ResourceData, meta interface{})
...
+			} else {
+				return err
+			}
 		} else {
-			if len(response.Connections) < 1 {
-				return fmt.Errorf("Error, failed to create instance because the network doesn't have at least 1 private services connection. Please see https://cloud.google.com/sql/docs/mysql/private-ip#network_requirements for how to create this connection.")
+			response, err := config.NewServiceNetworkingClient(userAgent).Services.Connections.List("services/servicenetworking.googleapis.com").Network(serviceNetworkingNetworkName).Do()
+			if err != nil {
+				// It is possible that the actor creating the SQL Instance might not have permissions to call servicenetworking.services.connections.list
+				log.Printf("[WARNING] Failed to list Service Networking of the project. Skipping Service Networking check.")
```
Comment (ScottSuarez) [inline third_party/terraform/resources/resource_sql_database_instance.go.erb:686]: Suggestion to move some of these nested `if` `else` blocks to functions. There are so many layers it can be hard to follow.

File: third_party/terraform/resources/resource_sql_database_instance.go.erb [hunk 3]
```diff
@@ -664,22 +664,30 @@ func resourceSqlDatabaseInstanceCreate(d *schema.ResourceData, meta interface{})
...
-			return err
-		}
-		response, err := config.NewServiceNetworkingClient(userAgent).Services.Connections.List("services/servicenetworking.googleapis.com").
-			Network(serviceNetworkingNetworkName).Do()
-		if err != nil {
-			// It is possible that the identity creating the SQL Instance might not have permissions to call servicenetworking.services.connections.list
-			log.Printf("[WARNING] Failed to list Service Networking of the project. Skipping Service Networking check.")
+			var gerr *googleapi.Error
+			if errors.As(err, &gerr) {
+				log.Printf("[DEBUG] retrieved googleapi error while creating sn name for %q. skipping precheck. code %v and message: %s", network, gerr.Code, gerr.Body)
```
Comment (ScottSuarez) [inline third_party/terraform/resources/resource_sql_database_instance.go.erb:678]: nit. `skipping precheck` but at this point we have already not decided to do the precheck as we received an error. `precheck skipped` instead? Mostly for the programmer rather than the user.

**Review by ScottSuarez — APPROVED** [review]
few comments surrounding readability

File: third_party/terraform/resources/resource_sql_database_instance.go.erb (hunk 1 above)
Comment (rileykarson) [reply third_party/terraform/resources/resource_sql_database_instance.go.erb:676]: This isn't entirely true! googleapi errors are effectively any error coming from the API client (assuming the base request is successful, otherwise it will be another type). We could filter on permissions errors, but it isn't worth doing so imo (the more fine-grained the filtering, the easier it is to get wrong)

File: third_party/terraform/resources/resource_sql_database_instance.go.erb (hunk 3 above)
Comment (rileykarson) [reply third_party/terraform/resources/resource_sql_database_instance.go.erb]: Done

File: third_party/terraform/resources/resource_sql_database_instance.go.erb (hunk 2 above)
Comment (rileykarson) [reply third_party/terraform/resources/resource_sql_database_instance.go.erb]: done
